### DISCUSSION TITLE
Redis Sentinel Support

### DISCUSSION THREAD

**Discussion by heilerich** [outline/outline#2608] [Ideas]
Hi there 👋🏻 

We would like to use [Redis Sentinel](https://redis.io/topics/sentinel) for better fault tolerance. I have already tested this and it seems to work quite well. The only changes that were required are basically checking if the Redis URL starts with `redis+sentinel://` (or something similar) and then passing a few more options to the `ioredis` instance (see [ioredis docs](https://github.com/luin/ioredis#sentinel))

Would you be open to merging this if I provide a PR?

Greetings

**Comment by tommoor** [comment]
Yes, this would be a great addition

**Comment by ToshY** [comment]
@heilerich Did you make a PR?

If not, maybe transfer this to issues and tag as enhancement?

**Comment by heilerich** [comment]
This was implemented in #3365

**Reply by ToshY** [reply]
@heilerich Thanks. I'm just confused why the [documentation ](https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e) doesn't include a single mention of sentinel 😕

**Reply by heilerich** [reply]
Well, the documentation doesn't really seem to care much about redis at all :) I think most people run outline in very small to mid sized deployments that don't have very strict availability constraints. So I guess anything other than a single user, one-instance redis installation could be considered an edge case.

Anyways, the sample environment file mentions support for advanced ioredis configuration here https://github.com/outline/outline/blob/34b586724b9570cc8912725d717135c28da2bc66/.env.sample#L24-L29

All the available options can be found in the [ioredis API reference](https://luin.github.io/ioredis/)
